src/js_native_api_v8.cc

```diff
@@ -1362,6 +1362,48 @@ napi_status napi_define_properties(napi_env env,
   return GET_RETURN_STATUS(env);
 }
 
+napi_status napi_object_freeze(napi_env env,
+                               napi_value object,
+                               napi_value* result) {
+  NAPI_PREAMBLE(env);
+
+  v8::Local<v8::Context> context = env->context();
+  v8::Local<v8::Object> obj;
+
+  CHECK_TO_OBJECT(env, context, obj, object);
+
+  v8::Maybe<bool> set_frozen =
+    obj->SetIntegrityLevel(context, v8::IntegrityLevel::kFrozen);
+
+  RETURN_STATUS_IF_FALSE(env, set_frozen.FromJust(), napi_generic_failure);
+
+  if (set_frozen.FromJust())
+    *result = v8impl::JsValueFromV8LocalValue(obj);
```

Comment:
what happens to the result if the freezing fails? should such scenario be documented?